ext/mysqli/mysqli_api.c

```diff
@@ -819,15 +819,52 @@ PHP_FUNCTION(mysqli_stmt_execute)
 {
 	MY_STMT		*stmt;
 	zval		*mysql_stmt;
+	zval		*input_params = NULL;
 #ifndef MYSQLI_USE_MYSQLND
 	unsigned int	i;
 #endif
 
-	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O", &mysql_stmt, mysqli_stmt_class_entry) == FAILURE) {
+	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O|a!", &mysql_stmt, mysqli_stmt_class_entry, &input_params) == FAILURE) {
```

Comment:
```suggestion
	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O|h!", &mysql_stmt, mysqli_stmt_class_entry, &input_params) == FAILURE) {
```
And then change `zval *input_params = NULL;` to `HashTable *input_params = NULL;` and replace usages of `Z_ARRVAL_P(input_params)` with just `input_params`.